components/style/servo/restyle_damage.rs

```diff
@@ -175,119 +175,32 @@ impl fmt::Display for ServoRestyleDamage {
-                            [ServoRestyleDamage::REPAINT, ServoRestyleDamage::REPOSITION,
-                            ServoRestyleDamage::STORE_OVERFLOW, ServoRestyleDamage::REFLOW_OUT_OF_FLOW],
-        [get_position.top, get_position.left,
-        get_position.right, get_position.bottom,
-        get_effects.opacity,
-        get_box.transform, get_box.transform_style, get_box.transform_origin,
-        get_box.perspective, get_box.perspective_origin
-    ]) || add_if_not_equal!(old, new, damage,
-                            [ServoRestyleDamage::REPAINT], [
-        get_color.color, get_background.background_color,
-        get_background.background_image, get_background.background_position_x,
-        get_background.background_position_y, get_background.background_repeat,
-        get_background.background_attachment, get_background.background_clip,
-        get_background.background_origin, get_background.background_size,
-        get_border.border_top_color, get_border.border_right_color,
-        get_border.border_bottom_color, get_border.border_left_color,
-        get_border.border_top_style, get_border.border_right_style,
-        get_border.border_bottom_style, get_border.border_left_style,
-        get_border.border_top_left_radius, get_border.border_top_right_radius,
-        get_border.border_bottom_left_radius, get_border.border_bottom_right_radius,
-        get_position.z_index, get_box._servo_overflow_clip_box,
-        get_inheritedtext.text_decorations_in_effect,
-        get_pointing.cursor, get_pointing.pointer_events,
-        get_effects.box_shadow, get_effects.clip, get_inheritedtext.text_shadow, get_effects.filter,
-        get_effects.mix_blend_mode, get_inheritedbox.image_rendering,
-
-        // Note: May require REFLOW et al. if `visibility: collapse` is implemented.
-        get_inheritedbox.visibility
-    ]);
+    restyle_damage_rebuild_and_reflow!(old, new, damage,
+            [ServoRestyleDamage::REPAINT, ServoRestyleDamage::REPOSITION,
+            ServoRestyleDamage::STORE_OVERFLOW, ServoRestyleDamage::BUBBLE_ISIZES,
+            ServoRestyleDamage::REFLOW_OUT_OF_FLOW, ServoRestyleDamage::REFLOW,
+            ServoRestyleDamage::RECONSTRUCT_FLOW]) ||
+        (new.get_box().display == Display::Inline &&
+         restyle_damage_rebuild_and_reflow_inline!(old, new, damage,
+            [ServoRestyleDamage::REPAINT, ServoRestyleDamage::REPOSITION,
+            ServoRestyleDamage::STORE_OVERFLOW, ServoRestyleDamage::BUBBLE_ISIZES,
+            ServoRestyleDamage::REFLOW_OUT_OF_FLOW, ServoRestyleDamage::REFLOW,
+            ServoRestyleDamage::RECONSTRUCT_FLOW])) ||
+    restyle_damage_reflow_and_bubble!(old, new, damage,
+            [ServoRestyleDamage::REPAINT, ServoRestyleDamage::REPOSITION,
+            ServoRestyleDamage::STORE_OVERFLOW, ServoRestyleDamage::BUBBLE_ISIZES,
+            ServoRestyleDamage::REFLOW_OUT_OF_FLOW, ServoRestyleDamage::REFLOW]) ||
+    restyle_damage_reflow!(old, new, damage,
```

Comment:
On second thought, this should be named `reflow_out_of_flow` instead of just `reflow`, because it doesn't include `REFLOW` damage.

`reflow_and_bubble` could be renamed to just `reflow`.